### DISCUSSION TITLE
Modifying records in the table

### DISCUSSION THREAD

**Discussion by abrzoza** [serenity-is/Serenity#6398] [Help]
Hi,
I'm starting with Serenity. I want to do a table lookup on the changeSelect2 event. I want to modify the searched records and save the modified ones in the database.
Please give me a hint on the best way to do this. Maybe someone would point out a link to an example or a piece of documentation.

**Comment by wirble** [comment]
You could do something like in your xxxDialog.ts file:

```
     this.form.YourRowField.changeSelect2(e => {
                //do something here on field changes
               //if you have id of the row, you can use the built in getlookup to get by id
               //if you need to do a custom lookup, you will need to build your own service endpoint, then call it....
               var fullName =  YourcustomRow.getLookup().itemById[row.Id].YourPersonFullName; 
      });

**Reply by abrzoza** [reply]
Thank you for the information.
I have a different goal. In the changeSelect2 event, I want to search for records from a table and modify them.

**Comment by minhhungit** [comment]
@abrzoza I think you can do like this:
- create a form with the select2 dropdown (this dropdown load data from table you want to modify), after clicking the dropdown (using changeselect2 event) it will call Retrieve service and load data to other controls on FORM for example textboxs
- the form will have a save button
- after modifing values then hit save button (call Save service)

![image](https://user-images.githubusercontent.com/2279508/170954926-fa25d082-eba0-4e61-89ca-39064e3edf1b.png)
